From 75491fb360e509cfd6abff8b32e0317257663858 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Thu, 11 Oct 2018 16:56:35 +0800
Subject: [PATCH 4816/5242] MLK-19888 gpu: imx: dpu: common: Use stream1 as
 the master stream for i.MX8QM

commit  8dc2e0b9aac6f24c21d2cc6c54928f4076da0e22 from
https://source.codeaurora.org/external/imx/linux-imx.git

Pixel combiner uses two display streams to drive one single display.
Either one of the two streams can be master stream and the other
slave stream.  If we use stream0 as the master stream, the overlay
covers the two streams could be unsynchronized sometimes. It looks
like there is one frame lag between the two streams.  However, it
turns out that using stream1 as the master stream can workaround
this issue.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-common.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/imx/dpu/dpu-common.c b/drivers/gpu/imx/dpu/dpu-common.c
index eec5a22..4a1e78b 100644
--- a/drivers/gpu/imx/dpu/dpu-common.c
+++ b/drivers/gpu/imx/dpu/dpu-common.c
@@ -605,7 +605,7 @@ static inline void dpu_cm_write(struct dpu_soc *dpu, u32 value,
 	.has_syncmode_fixup = true,
 	.syncmode_min_prate = 300000,
 	.singlemode_max_width = 1920,
-	.master_stream_id = 0,
+	.master_stream_id = 1,
 	.pixel_link_quirks = true,
 	.pixel_link_nhvsync = true,
 	.version = DPU_V2,
-- 
1.7.9.5

